src/Router.py

```diff
@@ -0,0 +1,158 @@
+        check if the 2 ips are from the same subnet of a router's interface and NOT the router's interface ip
+        '''
+        return False
+
+
+    def print_arp(self, arp_detailed, ethernet_detailed, ethertype):
+        print("****************_ETHERNET_FRAME_****************")
+        print("Dest MAC:        ", binascii.hexlify(ethernet_detailed[0]))
+        print("Source MAC:      ", binascii.hexlify(ethernet_detailed[1]))
+        print("Type:            ", binascii.hexlify(ethertype))
+        print("************************************************")
+        print("******************_ARP_HEADER_******************")
+        print("Hardware type:   ", binascii.hexlify(arp_detailed[0]))
+        print("Protocol type:   ", binascii.hexlify(arp_detailed[1]))
+        print("Hardware size:   ", binascii.hexlify(arp_detailed[2]))
+        print("Protocol size:   ", binascii.hexlify(arp_detailed[3]))
+        print("Opcode:          ", binascii.hexlify(arp_detailed[4]))
+        print("Source MAC:      ", binascii.hexlify(arp_detailed[5]))
+        print("Source IP:       ", socket.inet_ntoa(arp_detailed[6]))
+        print("Dest MAC:        ", binascii.hexlify(arp_detailed[7]))
+        print("Dest IP:         ", socket.inet_ntoa(arp_detailed[8]))
+        print("*************************************************\n")
+
+    def print_ping(self, ethertype, ethernet_detailed, ip_detailed, icmp_detailed):
+        print("****************_ETHERNET_FRAME_****************")
+        print("Dest MAC:        ", binascii.hexlify(ethernet_detailed[0]))
+        print("Source MAC:      ", binascii.hexlify(ethernet_detailed[1]))
+        print("Type:            ", binascii.hexlify(ethertype))
+        print("************************************************")
+        print("******************_IP_HEADER_******************")
+        print("Version...TTL:   ", binascii.hexlify(ip_detailed[0]))
+        print("Protocol type:   ", binascii.hexlify(ip_detailed[1]))
+        print("Checksum:        ", binascii.hexlify(ip_detailed[2]))
+        print("Source IP:       ", socket.inet_ntoa(ip_detailed[3]))
+        print("Dest IP:         ", socket.inet_ntoa(ip_detailed[4]))
+        print("*************************************************\n")
+        print("******************_ICMP_HEADER_******************")
+        print("Type:      ", binascii.hexlify(icmp_detailed[0]))
+        print("Code:      ", binascii.hexlify(icmp_detailed[1]))
+        print("Checksum:  ", binascii.hexlify(icmp_detailed[2]))
+        print("Rest:      ", binascii.hexlify(icmp_detailed[3]))
+        print("*************************************************\n")
+
+    def handle_arp(self, arp_packet, ethernet_detailed, ethertype, interface):
+        '''
+        Handles ARP requests by unpacking them and constructing a respose packet (if needded).
+        Prints the ARP packet contents.
+        '''
+        arp_header = arp_packet[0][14:42]       # ARP is 28 bytes payload
```

Comment:
There is a problem with this line, check [this](https://www.youtube.com/watch?v=f2Q_fO7fhJI) for further reference.